jannovar-core/src/main/java/de/charite/compbio/jannovar/reference/AminoAcidChangeNormalizer.java

```diff
@@ -21,46 +24,44 @@ public static AminoAcidChange truncateAltAfterStopCodon(AminoAcidChange change)
 	 *
-	 * @param ref    reference amino acid string to change
-	 * @param alt    alternative amino acid string to compare with the wild-type (ref)
-	 * @param change the {@link AminoAcidChange} to normalize
-	 * @return normalized AminoAcidChange
+	 * @param change the original amino acid change to be shifted
+	 * @param wtAASeq the wildtype amino acid sequence (i.e. the translated reference sequence)
+	 * @param varAASeq the predicted amino acid sequence induced by the variant
+	 * @return the amino acid change shifted to the first difference in the sequence, or the end of the ref/alt CDS
 	 */
-	public static AminoAcidChange normalizeDeletion(String ref, String alt, AminoAcidChange change) {
-		if (change.getRef().length() == 0)
-			return change;
-
-		// Compute shift of deletion.
-		int shift = 0;
-		while (change.getPos() + shift < ref.length() && change.getPos() + shift < alt.length()
-				&& ref.charAt(change.getPos() + shift) == alt.charAt(change.getPos() + shift))
-			shift += 1;
-		if (shift == 0)
-			return change;
-
-		// Build new AminoAcidChange.
-		StringBuilder changeRefBuilder = new StringBuilder();
-		changeRefBuilder.append(ref.substring(change.getPos() + shift, change.getPos() + shift + change.getRef().length()));
-		StringBuilder changeAltBuilder = new StringBuilder();
-		changeAltBuilder.append(alt.substring(change.getPos() + shift, change.getPos() + shift + change.getAlt().length()));
-		return new AminoAcidChange(change.getPos() + shift, changeRefBuilder.toString(), changeAltBuilder.toString());
+	public static AminoAcidChange shiftSynonymousChange(AminoAcidChange change, String wtAASeq, String varAASeq) {
+	  AminoAcidChange aminoAcidChange = truncateBothSides(change);
+		int position = aminoAcidChange.getPos();
+		int originalPosition = position;
+		int maxPosition = Math.min(
+				wtAASeq.length() - aminoAcidChange.getRef().length(),
+				varAASeq.length() - aminoAcidChange.getAlt().length());
+		while (position < maxPosition && wtAASeq.charAt(position) != '*'
+				&& wtAASeq.charAt(position) == varAASeq.charAt(position)) {
+			position++;
+		}
+		if (position == originalPosition) {
+			return aminoAcidChange;
+		}
+		return new AminoAcidChange(position,
+				wtAASeq.substring(position, position + aminoAcidChange.getRef().length()),
+				varAASeq.substring(position, position + aminoAcidChange.getAlt().length()));
 	}
 
 	/**
 	 * Truncate {@link AminoAcidChange} from both sides for matching ref/alt prefixes/suffixes.
 	 * <p>
 	 * Truncating of the prefixes is given higher priority to conform with the HGVS notation (you have to call
-	 * {@link #shiftInsertion}) afterwards.
+	 * {@link #shiftSynonymousChange}) afterwards.
 	 *
 	 * @param aaChange {@link AminoAcidChange} to truncate
 	 * @return updated {@link AminoAcidChange}
 	 */
-	public static AminoAcidChange truncateBothSides(AminoAcidChange aaChange) {
-		// TODO(holtgrem): Test me!
```

Comment:
I've removed this TODO since this is now a 'helper method' for `shiftSynonymousChange` and thus tested with it; I made sure to add additional test data that covers such cases to `AminoAcidChangeTest`.